include/pybind11/detail/internals.h

```diff
@@ -260,7 +260,25 @@ struct type_info {
 /// Each module locally stores a pointer to the `internals` data. The data
 /// itself is shared among modules with the same `PYBIND11_INTERNALS_ID`.
 inline internals **&get_internals_pp() {
+#if defined(PYPY_VERSION) || defined(GRAALVM_PYTHON) || PY_VERSION_HEX < 0x030C0000               \
+    || !defined(PYBIND11_SUBINTERPRETER_SUPPORT)
     static internals **internals_pp = nullptr;
+#else
+    static thread_local internals **internals_pp = nullptr;
+    // This is one per interpreter, we cache it but if the thread changed
```

Comment:
Could you please leave a terse comment here to explain why this needs to be `thread_local`? Maybe a reference to Python documentation, or `... a few words ... (see PR #5564 for details)`.

Reply:
Done